lms/static/scripts/ui-playground/components/ErrorComponents.js

```diff
@@ -42,29 +44,390 @@ function ErrorDialogExample() {
+                  displayed.
+                </li>
+                <li>
+                  For other types of JavaScript Error objects, or error-like
+                  objects, the <code>message</code> property of the Error, if
+                  present.
+                </li>
+              </ul>
+              <p>
+                When both a contextual hint and a message are shown, they are
+                shown separated by a colon, e.g.:
+              </p>
+              <pre>
+                <code>description: message</code>
+              </pre>
+              <p>
+                A message is <b>sometimes shown</b>.
+              </p>
+            </li>
+            <li>
+              <p>
+                Static canned <b>user-help text</b>
+              </p>
+              <p>
+                This provides basic links and help instructions and is{' '}
+                <b>always shown</b>.
+              </p>
+            </li>
+            <li>
+              <p>
+                Additional <b>error details</b> provided by our API
+              </p>
+              <p>
+                For errors from requests to our API, any <code>details</code> in
+                the response body will be JSON-stringified and shown in a
+                collapsed details section. Details are <b>sometimes shown.</b>
+              </p>
+            </li>
+          </ul>
+          <Library.Demo withSource title="With contextual hint and message">
             <ErrorDisplay
               error={fakeError}
-              description="This is a hard-coded description provided in the 'description' prop"
-            />
+              description="This is an app-provided contextual hint"
+            >
+              This is an example of prepared/canned text that can be rendered
+              depending on the error code at play.
+            </ErrorDisplay>
+          </Library.Demo>
+
+          <Library.Demo withSource title="Without contextual hint">
+            <ErrorDisplay error={fakeError}>
+              <p>
+                This is an example of prepared/canned text that can be rendered
+                depending on the error code at play.
+              </p>
+            </ErrorDisplay>
+          </Library.Demo>
+
+          <Library.Demo withSource title="Without contextual hint or message">
+            <ErrorDisplay error={{ details: { foo: 'bar' } }}>
+              <p>
+                This is an example of prepared/canned text that can be rendered
+                depending on the error code at play.
+              </p>
+            </ErrorDisplay>
+          </Library.Demo>
+
+          <Library.Demo
+            withSource
+            title="Without canned/error-code text or contextual hint"
+          >
+            <ErrorDisplay error={fakeError} />
+          </Library.Demo>
+        </Library.Example>
+      </Library.Pattern>
+      <Library.Pattern title="Error-mode errors">
```

Comment:
Each surrounding `Library.Pattern` sections refer to specific components, whereas this refers to several. It might help overall cohesiveness to split this into two separate `Library.Pattern`s, one for `ErrorDialogApp` and one for `OAuth2RedirectErrorApp`.